# Solutions Modélisation

#### Exercice 1

1. Déterminez la fonction de transfert lorsque l'entrée est la tension $u_{in}(t)$ et la sortie est $u_{out}(t)$

Le schéma électrique est rappelé ici:

<img src="./Images/Modelisation/circuit_double_rc.png" alt="Circuit double RC" width=500>

La fonction de transfert peut être obtenue par les lois des mailles et les relations tensions/courants sur chaque élément:

$$ 
  \left\{ \begin{array}{l}
    u_{in}(t) = R_1 (i_{C_1}(t) + i_{C_2}(t)) + R_2 i_{C_2}(t) + u_{out}(t) \\
    u_{C_1}(t) = R_2 i_{C_2}(t) + u_{out}(t) \\
    i_{C_1}(t) = C_1 \dfrac{du_{C_1}(t)}{dt} \\
    i_{C_2}(t) = C_2 \dfrac{du_{out}(t)}{dt}
  \end{array} \right.  
$$

En utilisant les relations tensions/courants, on obtient successivement:

\begin{align}
  u_{in}(t) &= R_1 \left(C_1 \dfrac{du_{C_1}(t)}{dt} + C_2 \dfrac{du_{out}(t)}{dt}\right) + R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \\
  &= R_1 \left(C_1 \dfrac{d\left( R_2 i_{C_2}(t) + u_{out}(t) \right)}{dt} + C_2 \dfrac{du_{out}(t)}{dt}\right) + R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \\
  &= R_1 \left(C_1 \dfrac{d\left( R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \right)}{dt} + C_2 \dfrac{du_{out}(t)}{dt}\right) + R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \\
  &= R_1 C_1 \dfrac{d\left( R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \right)}{dt} + R_1 C_2 \dfrac{du_{out}(t)}{dt} + R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \\
  &= R_1 C_1 \dfrac{d \left( R_2 C_2 \dfrac{du_{out}(t)}{dt} \right)}{dt} + R_1 C_1 \dfrac{d u_{out}(t)}{dt} + R_1 C_2 \dfrac{du_{out}(t)}{dt} + R_2 C_2 \dfrac{du_{out}(t)}{dt} + u_{out}(t) \\
  &= R_1 R_2 C_1 C_2 \dfrac{d^2 u_{out}(t)}{dt^2} + (R_1 C_1  + R_1 C_2 + R_2 C_2) \dfrac{du_{out}(t)}{dt} + u_{out}(t) \\
\end{align}

A ce stade, il ne reste qu'à passer dans le domaine de Laplace:

\begin{align}
  U_{in}(s) &= R_1 R_2 C_1 C_2 s^2 U_{out}(s) + (R_1 C_1  + R_1 C_2 + R_2 C_2) s U_{out}(s) + U_{out}(s) \\
  \frac{U_{out}(s)}{U_{in}(s)} &= \frac{1}{R_1 R_2 C_1 C_2 s^2 + (R_1 C_1  + R_1 C_2 + R_2 C_2) s + 1} \\
  H(s) &= \frac{1}{R_1 R_2 C_1 C_2 s^2 + (R_1 C_1  + R_1 C_2 + R_2 C_2) s + 1}
\end{align}

En remplaçant par les valeurs données, on a:

$$ H(s) = \frac{1}{2.4 s^2 + 9.1 s + 1} $$

---

2. Calculez la sortie $u_{out}(t)$ pour $u_{in}(t)$ égal à un échelon unité

Les pôles de la fonction de transfert peuvent être obtenus de différentes manières. Ce calcul étant élémentaire, utilisons une fonction pour l'obtenir:

In [4]:
import numpy as np

print(np.roots([2.4, 9.1, 1]))

[-3.67839255 -0.11327412]


L'expression de $U_{out}(s)$ est alors:

\begin{align}
  U_{out}(s) &= \frac{1}{2.4 (s + 3.6784) (s + 0.1133)} U_{in}(s) \\
  &= \frac{1}{2.4 (s + 3.6784) (s + 0.1133)} \frac{1}{s}
\end{align}

La décomposition en sommes de fractions simples doit donner:

$$ U_{out}(s) = \frac{1}{2.4 (s + 3.6784) (s + 0.1133) s} = \frac{A}{s + 3.6784} + \frac{B}{s + 0.1133} + \frac{C}{s} $$

Par le calcul des limites, on obtient:

\begin{align}
  A &= \lim_{s \rightarrow -3.6784} U_{out}(s) * (s + 3.6784) \\
  &= \lim_{s \rightarrow -3.6784} \frac{(s + 3.6784)}{2.4 (s + 3.6784) (s + 0.1133) s} \\
  &= \lim_{s \rightarrow -3.6784} \frac{1}{2.4 (s + 0.1133) s} \\
  &= 0.0318
\end{align}

\begin{align}
  B &= \lim_{s \rightarrow -0.1133} U_{out}(s) * (s + 0.1133) \\
  &= \lim_{s \rightarrow -0.1133} \frac{(s + 0.1133)}{2.4 (s + 3.6784) (s + 0.1133) s} \\
  &= \lim_{s \rightarrow -0.1133} \frac{1}{2.4 (s + 3.6784)} \\
  &= -1.0315
\end{align}

\begin{align}
  C &= \lim_{s \rightarrow 0} U_{out}(s) * s \\
  &= \lim_{s \rightarrow 0} \frac{s}{2.4 (s + 3.6784) (s + 0.1133) s} \\
  &= \lim_{s \rightarrow 0} \frac{1}{2.4 (s + 3.6784) (s + 0.1133)} \\
  &= 1
\end{align}

La décomposition donne:

$$ U_{out}(s) = \frac{0.0318}{s + 3.6784} + \frac{-1.0315}{s + 0.1133} + \frac{1}{s} $$

En passant par la transformée inverse, on obtient finalement:

$$ u_{out}(t) = 0.0318 e^{-3.6784 t} - 1.0315 e^{-0.1133 t} + 1 $$

---

#### Exercice 2

L'équation différentielle donne:

$$ M \ddot{x} = F - k x - b \dot{x} $$

Par la transformée de Laplace, on a:

\begin{align}
  M s^2 X(s) &= F(s) - k X(s) - b s X(s) \\
  \frac{X(s)}{F(s)} &= \frac{1}{M s^2 + b s + k}
\end{align}